fluster/utils.py

```diff
@@ -0,0 +1,15 @@
+def round_controlled(cycled_iterable, rounds=1):
+    """Raise StopIteration after <rounds> passes through a cycled iterable."""
+    round_start = None
+    rounds_completed = 0
+
+    for item in cycled_iterable:
+        if round_start is None:
+            round_start = item
+        elif item == round_start:
+            rounds_completed += 1
+
+        if rounds_completed == rounds:
+            raise StopIteration
+
+        yield item
```

Comment:
Big +1 to moving this into a helper function. It's useful, but not so useful it needs to be integral to the Cluster itself.